### PR TITLE
Fix infinite loop

### PR DESCRIPTION
**Pull request by annieztang**

### CODE REVIEW

File: src/components/UserMenu/UserMenu.js [hunk 1]
```diff
@@ -3,41 +3,44 @@ import { Joiner } from "../../helpers";
...
+            <UserMenuItem
+              key={user.userId}
+              user={user}
+              setSelected={setSelectedUser}
+              selected={selectedUser}
+            />
+          ))}
+        </div>
+      ) : (
+        <CircularProgress />
```
Comment (annieztang) [inline src/components/UserMenu/UserMenu.js:26]: added this

Comment (annieztang) [inline src/components/UserMenu/UserMenu.js:26]: added this

File: src/components/UserMenu/UserMenu.js [hunk 2]
```diff
@@ -3,41 +3,44 @@ import { Joiner } from "../../helpers";
...
+      className={Joiner(
+        "user-menu-item",
+        selected.userId === user.userId && "selected-user"
+      )}
+      onClick={() => setSelected(user)}
+    >
+      <UserIcon user={user} small />
+      <p>{user.username}</p>
+    </div>
+  );
```
Comment (annieztang) [inline src/components/UserMenu/UserMenu.js:45]: no change i think this is just formatting

File: src/components/pages/MessagePage.js [hunk 3]
```diff
@@ -5,111 +5,96 @@ import UsersList from "./../UsersList/UsersList";
...
-	const [messages, setMessages] = useState([]);
-	const currUsersGroups = getCurrGroups();
-	const [loading, setLoading] = useState(true);
-	const [delay] = useState(1000);
+  const [users, setUsers] = useState([]);
+  const [selectedUser, setSelectedUser] = useState({});
+  const [groups, setGroups] = useState(allGroups);
+  const [selectedGroup, setSelectedGroup] = useState(groups[0]);
+  const [messages, setMessages] = useState([]);
+  const currUsersGroups = getCurrGroups();
```
Comment (annieztang) [inline src/components/pages/MessagePage.js:20]: using dummy data for default value for groups

didn't need it for users
but for groups, something in the dialog required `selectedGroup` so it couldn't be an empty object

File: src/components/pages/MessagePage.js [hunk 4]
```diff
@@ -5,111 +5,96 @@ import UsersList from "./../UsersList/UsersList";
...
+    currUsersGroups,
+    getMessagesByGroup,
+    groups,
+    messages,
+    users,
+    setSelectedGroup,
+    selectedGroup,
+    setSelectedUser,
+    selectedUser,
+  };
```
Comment (annieztang) [inline src/components/pages/MessagePage.js:63]: Reordered so the context would contain the data from the api
i think...

File: src/components/pages/MessagePage.js [hunk 5]
```diff
@@ -5,111 +5,96 @@ import UsersList from "./../UsersList/UsersList";
...
+    const newCurrGroups = groups.filter(
+      (userGroup) => userGroup.user.userId === selectedUser.userId
+    );
+    if (
+      newCurrGroups.some(
+        (userGroup) => userGroup.group.groupId === selectedGroup.groupId
+      )
+    ) {
+      setSelectedGroup(newCurrGroups[0]);
+    }
```
Comment (annieztang) [inline src/components/pages/MessagePage.js:95]: .includes was nono

**Review by tylercodes1 — APPROVED** [review]

**Pull request merged**